gammapy/modeling/selection.py

```diff
@@ -0,0 +1,94 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from gammapy.stats.utils import sigma_to_ts
+from gammapy.modeling import Parameter, Fit
+
+
+class TestStatisticNested:
+    """Compute the test statistic (TS) between two nested hypothesis.
+    The null hypothesis is the minimal one, for which a set of parameters
+    are frozen to given values.
+
+    Parameters
+    ----------
+    parameters : `~gammapy.modeling.Parameters` or list of `~gammapy.modeling.Parameter`
+        List of parameters frozen for the null hypothesis.
+    null_values : list of float or `~gammapy.modeling.Parameters`
+        Values of the parameters frozen for the null hypothesis.
+        If a `Parameters` object or a list of `Parameters` is given
+        the null hypothesis follows the values of these parameters,
+        so this tests linked parameters versus unliked.
+    n_sigma : float
+        Threshold in number of sigma to switch from the null hypothesis
+        to the alternative one. Default is 2.
+    n_free_parameters : int
+        Number of free parameters to consider between the two hypothesis
+        in order to estimate the `ts_threshold`. Default is len(parameters).
+    fit : `Fit`
+        Fit instance specifying the backend and fit options.
+    """
+
+    def __init__(
+        self, parameters, null_values, n_sigma=2, n_free_parameters=None, fit=None
+    ):
+        self.parameters = parameters
+        self.null_values = null_values
+        self.n_sigma = n_sigma
+
+        if n_free_parameters is None:
+            n_free_parameters = len(parameters)
```

Comment:
In the docstring, it is mentioned that "parameters: List of parameters frozen for the null hypothesis". So its length is not the number of free parameters, is not it?

Reply:
I should clarify this in the docsting as : "parameters: List of parameters frozen for the null hypothesis but free for the test hypothesis".

Reply:
with, if this list contain frozen and not frozen, its length is still not the number of free parameters. We need instead `len(parameters.free_parameters)`, no?